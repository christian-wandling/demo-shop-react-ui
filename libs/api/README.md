# api

A comprehensive library for consuming the demo-shop api. Supports accessing a product catalog, user management,
shopping cart operations, and order processing.
