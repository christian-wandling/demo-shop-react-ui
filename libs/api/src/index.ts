export * from './lib/openapi';
export * from './lib/ApiContext';
export * from './lib/services/apiService';
export * from './lib/ApiProvider';
export * from './lib/services/apiService';
export * from './lib/middleware/authMiddleware';
