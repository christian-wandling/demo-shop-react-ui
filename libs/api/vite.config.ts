import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';

export default defineConfig(() => ({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/libs/api',
  plugins: [react()],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    watch: false,
    globals: true,
    environment: 'jsdom',
    reporters: ['default'],
    passWithNoTests: true,
    coverage: {
      reportsDirectory: './test-output/vitest/coverage',
      enabled: false,
    },
  },
}));
