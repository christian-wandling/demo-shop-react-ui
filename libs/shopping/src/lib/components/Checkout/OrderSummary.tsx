import { formatCurrency } from '@demo-shop-react-ui/shared';
import { CartItems } from '../CartItems/CartItems';
import { OrderSummaryProps } from '../../models/orderSummaryProps';

export const OrderSummary = ({
  items,
  totalPrice,
  onRemoveItem,
  onCheckout,
  checkoutButtonEnabled,
}: OrderSummaryProps) => {
  return (
    <section>
      <h2 className="text-lg font-medium text-gray-900" id="slide-over-title">
        Order Summary
      </h2>

      <CartItems items={items} onRemoveItem={onRemoveItem} />

      <div className="border-t border-gray-900/10 mt-12">
        <div className="mt-8 flex justify-between">
          <span>Total</span>
          <span>{formatCurrency(totalPrice)}</span>
        </div>
      </div>

      <section>
        <div className="mt-6 flex items-center justify-center gap-x-6">
          <button
            onClick={onCheckout}
            disabled={!checkoutButtonEnabled}
            className={`btn-checkout mt-6 lg:max-w-60 flex w-full items-center justify-center rounded-md border border-transparent px-8 py-3 text-base font-medium ${
              !checkoutButtonEnabled
                ? 'bg-gray-200 text-gray-800 pointer-events-none'
                : 'bg-indigo-600 text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2'
            }`}>
            Checkout
          </button>
        </div>
      </section>
    </section>
  );
};
