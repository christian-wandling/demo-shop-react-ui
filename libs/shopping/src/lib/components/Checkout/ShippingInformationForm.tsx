import { Form, FormInput, FormSelect } from '@demo-shop-react-ui/shared';
import { ShippingInformationProps } from '../../models/shippingInformationProps';
import { useEffect, useState } from 'react';
import { countrySelectOptions } from '../../constants/countrySelectOptions';
import { ShippingInformationFormData } from '../../models/shippingInformationFormSchema';

export const ShippingInformationForm = ({ methods, onUpdateUser }: ShippingInformationProps) => {
  const BREAKPOINT_MD = 768;
  const [shippingExtended, setShippingExtended] = useState(false);

  useEffect(() => {
    setShippingExtended(window.innerWidth >= BREAKPOINT_MD);

    const handleResize = () => {
      setShippingExtended(window.innerWidth >= BREAKPOINT_MD);
    };

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  const toggleShippingExtended = () => {
    setShippingExtended(!shippingExtended);
  };

  const updateButtonEnabled = methods.formState.isValid && methods.formState.isDirty;

  return (
    <Form<ShippingInformationFormData> methods={methods} onSubmit={onUpdateUser} className="user-form">
      <div className="space-y-8">
        <div
          className="flex justify-between items-center p-4 bg-gray-100 cursor-pointer lg:pointer-events-none"
          onClick={toggleShippingExtended}>
          <h2 className="text-base font-semibold leading-7 text-gray-900">Shipping information</h2>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className={`lg:hidden w-5 h-5 transition-transform duration-300 ${shippingExtended ? 'rotate-180' : ''}`}
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor">
            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M19 9l-7 7-7-7" />
          </svg>
        </div>
        <div
          className={`mt-4 lg:max-h-[1000px] grid md:grid-cols-8 gap-x-6 gap-y-4 transition-all duration-500 ease-in-out overflow-hidden ${
            shippingExtended ? 'max-h-[1000px]' : 'max-h-0'
          }`}>
          <FormInput
            name="firstname"
            label="Firstname"
            disabled
            required
            className="md:col-span-4"
            data-testid="checkout-form-element-firstname"
          />
          <FormInput
            name="lastname"
            label="Firstname"
            disabled
            required
            className="md:col-span-4"
            data-testid="checkout-form-element-lastname"
          />
          <FormInput
            name="email"
            label="Email"
            disabled
            required
            className="md:col-span-4"
            data-testid="checkout-form-element-email"
          />
          <FormInput name="phone" label="Phone" className="md:col-span-4" data-testid="checkout-form-element-phone" />
          <FormSelect
            name="address.country"
            label="Country"
            options={countrySelectOptions}
            required
            className="md:col-span-4"
            data-testid="checkout-form-element-country"
          />
          <FormInput
            name="address.street"
            label="Street"
            required
            className="md:col-span-5"
            data-testid="checkout-form-element-street"
          />
          <FormInput
            name="address.apartment"
            label="Apartment"
            required
            className="md:col-span-3"
            data-testid="checkout-form-element-apartment"
          />
          <FormInput
            name="address.city"
            label="City"
            required
            className="md:col-span-3"
            data-testid="checkout-form-element-city"
          />
          <FormInput
            name="address.region"
            label="Region"
            className="md:col-span-3"
            data-testid="checkout-form-element-region"
          />
          <FormInput
            name="address.zip"
            label="Zip"
            required
            className="md:col-span-2"
            data-testid="checkout-form-element-zip"
          />

          <button
            disabled={!updateButtonEnabled}
            className={`btn-update md:col-span-full float-right m-2 mt-6 lg:max-w-60 flex flex-1 items-center justify-center rounded-md border border-transparent px-8 py-3 text-base font-medium ${
              !updateButtonEnabled
                ? 'bg-gray-200 text-gray-800 pointer-events-none'
                : 'bg-indigo-600 text-white hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2'
            }`}
            data-testid="checkout-update-shipping-information-button">
            Update
          </button>
        </div>
      </div>
    </Form>
  );
};
