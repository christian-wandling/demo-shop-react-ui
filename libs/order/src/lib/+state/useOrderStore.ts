import { create } from 'zustand';
import { getApi, OrderResponse, OrderStatus } from '@demo-shop-react-ui/api';

interface OrderState {
  orders: OrderResponse[];

  getSortedOrders: () => OrderResponse[];
  getOrderById: (id: number) => OrderResponse | undefined;
  fetchOrders: () => Promise<void>;
  fetchOrderById: (id: number) => Promise<void>;
}

export const useOrderStore = create<OrderState>()((set, get) => {
  return {
    orders: [],

    fetchOrders: async () => {
      const { orderApi } = getApi();
      const res = await orderApi.getAllOrdersOfCurrentUser();
      set({ orders: res.items });
    },

    fetchOrderById: async (id: number) => {
      const { orderApi } = getApi();
      const order = await orderApi.getOrderById({ id });
      set(state => ({ orders: [...state.orders, order] }));
    },

    getSortedOrders: () => {
      const { orders } = get();

      return orders.sort((a, b) => {
        if (a.status === b.status) {
          return a.created > b.created ? -1 : 1;
        }

        return a.status === OrderStatus.Created ? -1 : 1;
      });
    },

    getOrderById: id => {
      return get().orders.find(order => order.id === id);
    },
  };
});
