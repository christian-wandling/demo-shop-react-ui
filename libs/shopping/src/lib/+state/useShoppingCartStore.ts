import { create } from 'zustand';
import { AddCartItemRequest, CartItemResponse, getApi, UpdateCartItemQuantityRequest } from '@demo-shop-react-ui/api';

interface ShoppingCartState {
  items: CartItemResponse[];
  sessionId: number | null;
  showCart: boolean;
  fetchCurrentSession: () => Promise<void>;
  add: (addCartItemRequest: AddCartItemRequest) => Promise<void>;
  remove: (id: number) => Promise<void>;
  updateItemQuantity: (id: number, updateCartItemQuantityRequest: UpdateCartItemQuantityRequest) => Promise<void>;
  getItemById: (id: number) => CartItemResponse | undefined;
  getItemByProductId: (productId: number) => CartItemResponse | undefined;
  setShowCart: (showCart: boolean) => void;

  getTotalPrice: () => number;

  getItemCount: () => number;

  hasActiveSession: () => boolean;
}

export const useShoppingCartStore = create<ShoppingCartState>()((set, get) => {
  return {
    items: [],
    sessionId: null,
    showCart: false,

    getTotalPrice(): number {
      return get().items.reduce((sum, item) => sum + item.totalPrice, 0);
    },

    getItemCount(): number {
      return get().items.reduce((sum, item) => sum + item.quantity, 0);
    },

    hasActiveSession(): boolean {
      return get().sessionId !== null;
    },

    fetchCurrentSession: async () => {
      const { shoppingSessionApi } = getApi();
      const session = await shoppingSessionApi.resolveCurrentShoppingSession();
      set({ sessionId: session.id, items: session.items });
    },

    add: async (addCartItemRequest: AddCartItemRequest) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const itemInCart = get().getItemByProductId(addCartItemRequest.productId);

      if (itemInCart) {
        get().updateItemQuantity(itemInCart.id, { quantity: itemInCart.quantity + 1 });
        return;
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.addCartItem({ addCartItemRequest });
      get().fetchCurrentSession();
    },

    remove: async (id: number) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const itemInCart = get().getItemById(id);

      if (!itemInCart) {
        throw new Error('Item not found');
      }

      if (itemInCart.quantity > 1) {
        get().updateItemQuantity(itemInCart.id, { quantity: itemInCart.quantity - 1 });
        return;
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.removeCartItem({ id });
      get().fetchCurrentSession();
    },

    updateItemQuantity: async (id: number, updateCartItemQuantityRequest: UpdateCartItemQuantityRequest) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.updateCartItemQuantity({ id, updateCartItemQuantityRequest });
      get().fetchCurrentSession();
    },

    getItemById: id => {
      return get().items.find(item => item.id === id);
    },

    getItemByProductId: productId => {
      return get().items.find(item => item.productId === productId);
    },

    setShowCart: (showCart: boolean) => {
      set(state => ({ ...state, showCart }));
    },
  };
});
