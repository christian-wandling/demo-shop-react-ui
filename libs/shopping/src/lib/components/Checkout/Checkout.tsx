import { useEffect } from 'react';
import { useForm, UseFormReturn } from 'react-hook-form';
import { useNavigate } from 'react-router';
import { ShippingInformationForm } from './ShippingInformationForm';
import { useShoppingCartStore } from '../../+state/useShoppingCartStore';
import { useUserStore } from '@demo-shop-react-ui/user';
import { OrderSummary } from './OrderSummary';
import { ShippingInformationFormData, shippingInformationFormSchema } from '../../models/shippingInformationFormSchema';
import { zodResolver } from '@hookform/resolvers/zod';

export default function Checkout() {
  const items = useShoppingCartStore(state => state.items);
  const getTotalPrice = useShoppingCartStore(state => state.getTotalPrice);
  const checkout = useShoppingCartStore(state => state.checkout);
  const removeItem = useShoppingCartStore(state => state.remove);
  const user = useUserStore(state => state.user);
  const updateUserAddress = useUserStore(state => state.updateUserAddress);
  const updateUserPhone = useUserStore(state => state.updateUserPhone);
  const navigate = useNavigate();

  const methods: UseFormReturn<ShippingInformationFormData> = useForm<ShippingInformationFormData>({
    mode: 'onBlur',
    resolver: zodResolver(shippingInformationFormSchema),
    defaultValues: {
      firstname: user?.firstname || '',
      lastname: user?.lastname || '',
      email: user?.email || '',
      phone: user?.phone || '',
      address: {
        country: user?.address?.country || '',
        street: user?.address?.street || '',
        apartment: user?.address?.apartment || '',
        city: user?.address?.city || '',
        region: user?.address?.region || '',
        zip: user?.address?.zip || '',
      },
    },
  });

  const checkoutButtonEnabled = items.length > 0 && !!user?.address && !methods.formState.isDirty;

  useEffect(() => {
    if (user) {
      methods.reset({
        firstname: user.firstname || '',
        lastname: user.lastname || '',
        email: user.email || '',
        phone: user.phone || '',
        address: {
          country: user.address?.country || '',
          street: user.address?.street || '',
          apartment: user.address?.apartment || '',
          city: user.address?.city || '',
          region: user.address?.region || '',
          zip: user.address?.zip || '',
        },
      });
    }
  }, [methods, user]);

  const updateUser = async () => {
    const formData = methods.getValues();
    const formState = methods.formState;

    if (formState.isDirty) {
      if (formState.dirtyFields.address && formData.address) {
        await updateUserAddress(formData.address);
      }

      if (formState.dirtyFields.phone) {
        await updateUserPhone({ phone: formData.phone });
      }
    }
  };

  const handleCheckout = async () => {
    try {
      await checkout();
      navigate('/products');
    } catch (err) {
      console.error('Checkout failed:', err);
    }
  };

  return (
    <div className="bg-gray-100 sm:m-2 lg:m-6 p-4 sm:p-8 lg:p-10 sm:rounded" data-testid="checkout">
      <div className="flex flex-col lg:flex-row justify-between gap-8">
        <ShippingInformationForm methods={methods} onUpdateUser={updateUser} />
        <OrderSummary
          items={items}
          totalPrice={getTotalPrice()}
          checkoutButtonEnabled={checkoutButtonEnabled}
          onRemoveItem={removeItem}
          onCheckout={handleCheckout}
        />
      </div>
    </div>
  );
}
