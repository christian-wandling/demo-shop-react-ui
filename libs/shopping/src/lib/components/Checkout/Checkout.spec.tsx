import { beforeEach, describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Checkout } from './Checkout';
import { useShoppingCartStore } from '../../+state/useShoppingCartStore';
import { CartItemResponse, UserResponse } from '@demo-shop-react-ui/api';
import { useUserStore } from '@demo-shop-react-ui/user';
import { useForm } from 'react-hook-form';

const navigateMock = vi.fn();

vi.mock('react-router', () => ({
  useNavigate: () => navigateMock,
}));

vi.mock('react-hook-form', () => ({
  useForm: vi.fn().mockReturnValue({
    reset: vi.fn(),
    getValues: vi.fn().mockReturnValue({
      id: 1,
      firstname: 'John',
      lastname: 'Doe',
      email: 'john.doe@example.com',
      phone: '123456789',
      address: {
        country: 'USA',
        street: 'Main St',
        apartment: 'Apt 1',
        city: 'New York',
        region: 'NY',
        zip: '10001',
      },
    }),
    formState: {
      isDirty: false,
      dirtyFields: {},
    },
  } as any),
}));

const mockItems: CartItemResponse[] = [
  {
    id: 1,
    productName: 'Product 1',
    productThumbnail: 'productThumbnail1',
    unitPrice: 10,
    quantity: 1,
    totalPrice: 10,
    productId: 1,
  },
  {
    id: 2,
    productName: 'Product 2',
    productThumbnail: 'productThumbnail2',
    unitPrice: 20,
    quantity: 2,
    totalPrice: 40,
    productId: 2,
  },
];

const checkout = vi.fn();
const remove = vi.fn();

const mockShoppingCartState = {
  items: mockItems,
  getTotalPrice: vi.fn(),
  checkout,
  remove,
};

vi.mock('../../+state/useShoppingCartStore', () => ({
  useShoppingCartStore: vi.fn(selectorOrState => {
    if (typeof selectorOrState === 'function') {
      return selectorOrState(mockShoppingCartState);
    }
    return mockShoppingCartState;
  }),
}));

const updateUserAddress = vi.fn();
const updateUserPhone = vi.fn();

const mockUser: UserResponse = {
  id: 1,
  firstname: 'John',
  lastname: 'Doe',
  email: 'john.doe@example.com',
  phone: '123456789',
  address: {
    country: 'USA',
    street: 'Main St',
    apartment: 'Apt 1',
    city: 'New York',
    region: 'NY',
    zip: '10001',
  },
};

const mockUserState = {
  user: mockUser,
  updateUserAddress,
  updateUserPhone,
};

vi.mock('@demo-shop-react-ui/user', () => ({
  useUserStore: vi.fn(selectorOrState => {
    if (typeof selectorOrState === 'function') {
      return selectorOrState(mockUserState);
    }
    return mockUserState;
  }),
}));

vi.mock('./ShippingInformationForm', () => ({
  ShippingInformationForm: ({ methods, onUpdateUser }: any) => (
    <div data-testid="shipping-form">
      <button data-testid="update-user-button" onClick={onUpdateUser}>
        Update User
      </button>
      <span>Is Form Dirty: {methods.formState.isDirty.toString()}</span>
      <span>Is Form Valid: {!methods.formState.isInvalid}</span>
    </div>
  ),
}));

vi.mock('./OrderSummary', () => ({
  OrderSummary: ({ items, totalPrice, checkoutButtonEnabled, onRemoveItem, onCheckout }: any) => (
    <div data-testid="order-summary">
      <span>Total Items: {items.length}</span>
      <span>Total Price: {totalPrice}</span>
      <button data-testid="checkout-button" disabled={!checkoutButtonEnabled} onClick={onCheckout}>
        Checkout
      </button>
      <button data-testid="remove-item-button" onClick={() => onRemoveItem(1)}>
        Remove Item
      </button>
    </div>
  ),
}));

describe('Checkout Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders the component with shipping form and order summary', () => {
    render(<Checkout />);

    expect(screen.getByTestId('shipping-form')).toBeTruthy();
    expect(screen.getByTestId('order-summary')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    useShoppingCartStore().items = mockItems;
    useUserStore().user = mockUser;

    const { baseElement } = render(<Checkout />);

    expect(baseElement).toMatchSnapshot();
  });

  it('enables checkout button when conditions are met', () => {
    render(<Checkout />);

    const checkoutButton: HTMLButtonElement = screen.getByTestId('checkout-button');
    expect(checkoutButton.disabled).toBeFalsy();
  });

  it('updates user address when form is dirty', () => {
    const mockMethods = {
      reset: vi.fn(),
      getValues: vi.fn().mockReturnValue(mockUser),
      formState: {
        isDirty: true,
        dirtyFields: {
          address: {
            country: true,
            street: true,
            city: true,
            apartment: true,
            region: true,
            zip: true,
          },
        },
      },
    } as any;

    vi.mocked(useForm).mockReturnValueOnce(mockMethods);

    render(<Checkout />);

    const updateUserButton = screen.getByTestId('update-user-button');
    fireEvent.click(updateUserButton);

    expect(updateUserAddress).toHaveBeenCalledWith(mockUser.address);
    expect(updateUserPhone).not.toHaveBeenCalled();
  });

  it('updates user phone when form is dirty', () => {
    const mockMethods = {
      reset: vi.fn(),
      getValues: vi.fn().mockReturnValue(mockUser),
      formState: {
        isDirty: true,
        dirtyFields: {
          phone: true,
        },
      },
    } as any;

    vi.mocked(useForm).mockReturnValueOnce(mockMethods);

    render(<Checkout />);

    const updateUserButton = screen.getByTestId('update-user-button');
    fireEvent.click(updateUserButton);

    expect(updateUserPhone).toHaveBeenCalledWith({ phone: mockUser.phone });
    expect(updateUserAddress).not.toHaveBeenCalled();
  });

  it('calls checkout and navigates on successful checkout', async () => {
    render(<Checkout />);

    const checkoutButton = screen.getByTestId('checkout-button');
    fireEvent.click(checkoutButton);

    expect(checkout).toHaveBeenCalled();
    await waitFor(() => {
      expect(navigateMock).toHaveBeenCalledWith('/products');
    });
  });

  it('handles checkout errors gracefully', async () => {
    const consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {
      return;
    });
    checkout.mockRejectedValue(new Error('Checkout failed'));

    render(<Checkout />);

    const checkoutButton = screen.getByTestId('checkout-button');
    fireEvent.click(checkoutButton);

    expect(checkout).toHaveBeenCalled();
    await waitFor(() => {
      expect(consoleErrorSpy).toHaveBeenCalledWith('Checkout failed:', expect.any(Error));
      expect(navigateMock).not.toHaveBeenCalled();
    });

    consoleErrorSpy.mockRestore();
  });

  it('calls remove item when remove button is clicked', () => {
    render(<Checkout />);

    const removeItemButton = screen.getByTestId('remove-item-button');
    fireEvent.click(removeItemButton);

    expect(remove).toHaveBeenCalledWith(1);
  });

  it('resets form when user data changes', async () => {
    const spy = vi.spyOn(useForm(), 'reset');
    const updatedUser = { ...mockUser, phone: '555-444-333' };

    useUserStore().user = updatedUser;

    render(<Checkout />);

    expect(spy).toHaveBeenCalledWith({ ...updatedUser, id: undefined });
  });
});
