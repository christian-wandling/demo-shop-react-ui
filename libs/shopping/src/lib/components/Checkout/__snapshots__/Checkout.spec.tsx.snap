// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout Component > matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="bg-gray-100 sm:m-2 lg:m-6 p-4 sm:p-8 lg:p-10 sm:rounded"
    >
      <div
        class="flex flex-col lg:flex-row justify-between gap-8"
      >
        <div
          data-testid="shipping-form"
        >
          <button
            data-testid="update-user-button"
          >
            Update User
          </button>
          <span>
            Is Form Dirty: 
            false
          </span>
          <span>
            Is Form Valid: 
          </span>
        </div>
        <div
          data-testid="order-summary"
        >
          <span>
            Total Items: 
            2
          </span>
          <span>
            Total Price: 
          </span>
          <button
            data-testid="checkout-button"
          >
            Checkout
          </button>
          <button
            data-testid="remove-item-button"
          >
            Remove Item
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
